src/vos/tests/vts_dtx.c

```diff
@@ -746,7 +746,7 @@ dtx_17(void **state)
 
 	/* Commit the first 4 DTXs. */
 	rc = vos_dtx_commit(args->ctx.tc_co_hdl, xid, 4, NULL);
-	assert_int_equal(rc, 4);
+	assert_rc_equal(rc, 4);
```

Comment:
assert_int_equal()?

Reply:
so vos_dtx_commit doesn't return a DAOS retcode?

Reply:
I'm checking this note https://github.com/daos-stack/daos/blob/master/src/include/daos_srv/vos.h#L97 , so in this case this assert is checking that the commit count is 4, right? that's why it should be assert_int_equal rather than assert_rc_equal

Reply:
4 isn't a DAOS error code, but looking at the documentation for this function it returns negative DAOS return code on error, or positive number of operations, so this code is probably right.  I've not seen any DAOS functions that use this style before however.

Reply:
so I suppose if it doesn't return 4 then it'll be a DAOS errorcode, so you do want assert_rc_equal().  I just wasn't familar with the interface and I know that 4 isn't a errorcode.

Sorry for the noise.

Reply:
right, I agree. No problem, thanks for checking this patch!

Reply:
@ashleypittman yeah, there are cases where the macro fails which is why I asked to add the print_message there.  For example, if it returned 3, you would get DER_UNKNOWN != DER_UNKNOWN but the output would have 3 != 4 above it.

Since cmocka doesn't provide any means for printing a custom error that shows up in the XML, this seemed like the 99% solution to what I see is a really annoying problem